pkg/csi/service/node.go

```diff
@@ -394,7 +395,8 @@ func (driver *vsphereCSIDriver) NodeGetInfo(
 	var (
 		accessibleTopology map[string]string
 	)
-	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.ImprovedVolumeTopology) {
+	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.ImprovedVolumeTopology) ||
```

Comment:
So here we need to enable both feature gates? `tkgh-ha` and `improvedVolumeTopology`.

when feature gate for `tkgs-ha` is enabled, it should enable the feature completely without having depdedency on other feature. we added `improvedVolumeTopology` feature-gate for vanilla. Can we avoid having a dependency on other feature gates?

Reply:
No. To enter into this code block, we expected either `tkgh-ha` or `improve-volume-topology` is enabled. There is no such thing that one depends on the other. Additionally, `tkgh-ha` is only available in non-vanilla clusters while `improve-volume-topology` is only available in vanilla clusters, so these two FSSs are mutually exclusive.

Reply:
we need only one feature gate to guard changes

Reply:
Updated the change based on offline discussion